libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/analytics/UtdTracker.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.libraries.matrix.impl.analytics
+
+import im.vector.app.features.analytics.plan.Error
+import io.element.android.services.analytics.api.AnalyticsService
+import org.matrix.rustcomponents.sdk.UnableToDecryptDelegate
+import org.matrix.rustcomponents.sdk.UnableToDecryptInfo
+import timber.log.Timber
+import javax.inject.Inject
+
+class UtdTracker @Inject constructor(
+    private val analyticsService: AnalyticsService,
+) : UnableToDecryptDelegate {
+    override fun onUtd(info: UnableToDecryptInfo) {
+        Timber.d("onUtd for event ${info.eventId}, timeToDecryptMs: ${info.timeToDecryptMs}")
+        // TODO info will contain more information in the future, so that the app can report more precise data to the analytics.
+        val event = Error(
+            context = null,
+            cryptoModule = Error.CryptoModule.Rust,
+            domain = Error.Domain.E2EE,
+            // TODO get a more specific error name from `info`
+            name = Error.Name.OlmKeysNotSentError,
+        )
+        analyticsService.capture(event)
+    }
```

Comment:
I have spoken with @BillCarsonFr about this, and seems that we want to actually have the analytics event error include also a properties value, to that we can also log the `timeToDecryptMs` as a property (and pass it -1 in case is nil/null) but this might require a new analytics event release/update

Reply:
Created a PR on schema https://github.com/matrix-org/matrix-analytics-events/pull/92